addons/BLUAlert/BLUAlert.lua

```diff
@@ -0,0 +1,89 @@
+--[[
+BLUAlert v1.20131102
+
+Copyright (c) 2015, Kainsin
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+* Neither the name of BLUAlert nor the
+names of its contributors may be used to endorse or promote products
+derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Kainsin BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+require('sets')
+res = require('resources')
+chat = require('chat')
+
+_addon.name    = 'BLUAlert'
+_addon.author  = 'Kainsin'
+_addon.version = '1.0.0.0'
+
+-- Some BLU spells have a different name then the monster abilities they come from.
+blu_different_names = {
+    ["Everyone's Grudge"]     = "Evryone. Grudge",
+    ["Nature's Meditation"]   = "Nat. Meditation",
+    ["Orcish Counterstance"]  = "O. Counterstance",
+    ["Tempestuous Upheaval"]  = "Tem. Upheaval",
+    ["Atramentous Libations"] = "Atra. Libations",
+    ["Winds of Promyvion"]    = "Winds of Promy.",
+    ["Quadratic Continuum"]   = "Quad. Continuum",
+}
+
+-- Traverse through all of the BLU spells looking for the one with the given name.
+blu_spells = res.spells:type('BlueMagic')
+function find_blu_spell(monster_ability_name)
+    for i,v in pairs(blu_spells) do
+        if (v.en == monster_ability_name) then
+            return v.id
+        end
+    end
+end
+
+-- Since the action packet gives monster abilities by ID, we'll want to create a
+-- Monster Ability -> BLU Spell mapping to quickly find out which monster ability
+-- corresponds to which spell.
+spell_id_map = {}
+for i,v in pairs(res.monster_abilities) do
+    monster_ability_name = blu_different_names[v.en] or v.en
```

Comment:
While it doesn't matter much for such a small addon it's considered good form to mark local variables as locals (*Lua* defaults to global), so they don't leak into the surrounding scope, which could potentially cause issues. Simply change the line to this:
```lua
    local monster_ability_name = blu_different_names[v.en] or v.en
```

And same as above, use `english` instead of `en`.